inst/eDISH_app/modules/renderSettings/renderSettings.R

```diff
@@ -60,45 +60,72 @@ renderSettings <- function(input, output, session, data, settings, status){
   ######################################################################
 
   output$data_mapping_ui <- renderUI({
-    req(input$charts)
+    charts <- isolate(input$charts)
+    req(charts)
     tagList(
       createSettingsUI(
         data=data(),
         settings = settings(),
         setting_cat_val = "data",
-        charts=input$charts,
+        charts=charts,
         ns=ns
       )
     )
   })
 
 
   output$measure_settings_ui <- renderUI({
-    req(input$charts)
+    charts <- isolate(input$charts)
+    req(charts)
     tagList(
       createSettingsUI(
         data=data(),
         settings = settings(),
         setting_cat_val = "measure",
-        charts=input$charts,
+        charts=charts,
         ns=ns
       )
     )
   })
 
   output$appearance_settings_ui <- renderUI({
-    req(input$charts)
+    charts <- isolate(input$charts)
+    req(charts)
     tagList(
       createSettingsUI(
         data=data(),
         settings = settings(),
         setting_cat_val = "appearance",
-        charts=input$charts,
+        charts=charts,
         ns=ns
       )
     )
   })
 
+  ######### Hide Settings that are not relevant to selected charts ########
+  observeEvent(input$charts,{
```

Comment:
Good work!!  App crashes if no charts are selected, so we will need to accommodate that case here.

Reply:
Yeah, I noticed that - it was present before the change but definitely something that needs to be resolved